src/DynamoUtilities/SmartObservableCollection.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Linq;
+
+namespace DynamoUtilities
+{
+    /// <summary>
+    /// Wrapper over System.Collections.ObjectModel.ObservableCollection that fires minimal notifications.
+    /// This class supports batch operations that should defer CollectionChaned notifications.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class SmartObservableCollection<T> : ObservableCollection<T>
```

Comment:
why make this public?

Reply:
because it is used to bind to WPF controls